sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -159,19 +160,22 @@ class SessionCatalog(
   }
 
   private val tableRelationCache: Cache[QualifiedTableName, LogicalPlan] = {
-    var builder = CacheBuilder.newBuilder()
+    var builder = Caffeine.newBuilder()
       .maximumSize(cacheSize)
 
     if (cacheTTL > 0) {
       builder = builder.expireAfterWrite(cacheTTL, TimeUnit.SECONDS)
     }
-
-    builder.build[QualifiedTableName, LogicalPlan]()
+    // Wrapping as CaffeinatedGuava to be compatible with
+    // the get(key, valueLoader) API of Guava cache
+    builder.build()
   }
 
   /** This method provides a way to get a cached plan. */
   def getCachedPlan(t: QualifiedTableName, c: Callable[LogicalPlan]): LogicalPlan = {
-    tableRelationCache.get(t, c)
+    tableRelationCache.get(t, new JFunction[QualifiedTableName, LogicalPlan] {
```

Comment:
nit: maybe:
```scala
    tableRelationCache.get(t, (_: QualifiedTableName) => c.call())
```

Reply:
d69df8e fixed these comments. The change of here need to see if it can be compiled